### ISSUE TITLE
Unable to connect to server from host machine.

### ISSUE THREAD

**Issue by JXPrime** [jekyll/docker#91]
I'm new to Docker so it's probable I'm misunderstanding something.

I'm running Windows 10 and installed the Docker Toolbox. Clicking on the Docker QuickStart shell, and running the docker command seen in the screenshot below, I'm unable to connect to the server from the host machine.

![image](https://cloud.githubusercontent.com/assets/1841009/12997234/6153013e-d0f4-11e5-8611-b5c6f76fdfd2.png)

However, if I run a Vagrant VM and run Jekyll from inside there, I'm able to connect:

![image](https://cloud.githubusercontent.com/assets/1841009/12997220/37817174-d0f4-11e5-9a1d-588cdaaea111.png)

Is there some bit of network configuration I've missed/misundertsood with Docker?

**Comment by envygeeks** [comment]
What happens when you do: https://github.com/jekyll/docker#on-docker-machine?

**Comment by JXPrime** [comment]
What's the difference, for the uneducated like myself? :grin: 

![image](https://cloud.githubusercontent.com/assets/1841009/12997659/e8dc44f4-d0f8-11e5-8716-5c143666382b.png)

**Comment by envygeeks** [comment]
The difference that you should take out of all of that is the `docker-machine` stuff on the `-p`, Docker Toolbox uses Docker-Machine uses Virtualbox, you have to pull out your Virtualbox IP or let them get exposed entirely (as far as I'm aware.)

Don't blindly copy and paste that syntax, merge the `-p docker-machine` stuff (changing the port obviously :wink:) into your original command and see if you can get to the instance.

**Comment by envygeeks** [comment]
That said, if you aren't using Docker Machine -- through Docker Toolbox or otherwise, I don't know what's going on and this ticket will probably stalled until I can get onto a Windows machine, at which point I'll ask  for you to provide as much as you can so I can replicate it on my machines.

**Comment by JXPrime** [comment]
Yea, any command I try now is giving me that same error (presumably) from Jekyll about the Permission denied. :/ Should I not be browsing to localhost:4567, but instead the ip given by the `docker-machine ip` command?

**Comment by envygeeks** [comment]
@JXPrime I honestly don't know for Windows and I have very little experience with Docker-Machine, but I will find out this weekend and make sure our documentation is complete for Docker Machine.

**Comment by JXPrime** [comment]
No worries, I'm sure it's some bit of learning I missed along the way somewhere. :blush:

**Comment by JXPrime** [comment]
Something else I've noticed, is that none of the files on my local machine appear to exist inside the docker container...if you notice in my screenshot above, it says no configuration file was found, but there is one there. I've noticed also when doing the docker quickstarts, that the volume mapping has never worked for me.

**Comment by JXPrime** [comment]
I figured it out. :blush:

By default `docker-machine` mounts the home directory (so c:\users\foo) as available inside the docker machine. My source code was outside that directory. I cloned my repo into a folder inside my home directory, and was able to get the jekyll image to serve the content (I no longer receive the permission error as before). This, combined with my assumption from before (about accessing the host vm, rather than the host machine) equals success:

![image](https://cloud.githubusercontent.com/assets/1841009/13242433/c5740800-d9b1-11e5-84a7-40abc02a5f31.png)

Note I'm not browsing to `localhost` (my assumption before), but the ip of the docker-machine.
